src/main/resources/META-INF/plugin.xml

```diff
@@ -577,8 +577,13 @@
         <!-- Move Actions -->
 
         <moveLeftRightHandler language="Rust" implementationClass="org.rust.ide.actions.RsMoveLeftRightHandler"/>
-        <statementUpDownMover implementation="org.rust.ide.actions.RsFunctionUpDownMover"/>
-        <statementUpDownMover implementation="org.rust.ide.actions.RsStructItemUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsFunctionUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsStructItemUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsImplItemUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsTraitItemUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsModItemUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsMacroCallUpDownMover"/>
+        <statementUpDownMover implementation="org.rust.ide.actions.mover.RsMacroDefinitionUpDownMover"/>
```

Comment:
It seems to me that all these movers work similarly: they move items inside a module. So perhaps we could create only a single mover instead, which will handle all items?

Reply:
Yes. I need to find a nice solution to support easy extension for match pattern usw...